src/sql/azdata.proposed.d.ts

```diff
@@ -1523,6 +1531,13 @@ declare module 'azdata' {
 			 */
 			displayValue: string | number | boolean;
 		}
+
+		export interface Dictionary {
```

Comment:
this seems like a very generic interface to exporting from the azdata extensibility namespace.  I wonder if it's better to just declare the JS dictionary/array where this interface is used, or give this a more scoped name like `ExecutionPlanMetricsDictionary`.  Personally, I think we can use the JS type directly without loss of readability/etc.

Reply:
+1, we shouldn't be making our own version of stuff that already exists. You can just use `Record<string, string | undefined>` as the type. We probably don't want to use Dictionary since that doesn't transfer well over RPC.

Reply:
Just a question, but does `Dictionary` not transfer well over RPC because it collides with C#'s Dictionary type, or is it just that Dictionary's don't serialize well?